.github/workflows/integration_test.yml

```diff
@@ -1,43 +1,66 @@
-name: CI-integration-testing
+name: CI integration testing
 
 on:
-  [push, pull_request]
+  workflow_call:
+    secrets:
+      GHCR_USERNAME:
+        required: true
+      GHCR_TOKEN:
+        required: true
+      NARRATIVE_TEST_TOKEN:
+        required: true
 
 jobs:
-  build-and-test:
+  run_integration_tests:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - uses: conda-incubator/setup-miniconda@v2.0.0
+    -
+      name: Log in to the Container registry
+      uses: docker/login-action@v1
       with:
-        miniconda-version: 'latest'
-        activate-environment: test-environment
-        python-version: 3.6.10
-        auto-activate-base: false
-        auto-update-conda: true
-        condarc-file: test/condarc.yml
-
-    - name: Use Node JS 16.x
-      uses: actions/setup-node@v1
+        registry: ghcr.io
+        username: ${{ secrets.GHCR_USERNAME }}
+        password: ${{ secrets.GHCR_TOKEN }}
+    -
+      name: Repo checkout
+      uses: actions/checkout@v2
+    -
+      name: Set up environment
+      run: |
+        if [ ${{ github.base_ref }} == 'master' ]; then
+            branch=''
+        else
+            branch="-""${{ github.base_ref }}"
+        fi
+        narrative_version=`grep '\"version\":' src/config.json.templ | awk '{print $2}' | sed 's/"//g'`
+        narrative_git_hash=`grep '\"git_commit_hash\":' src/config.json.templ | awk '{print $2}' | sed 's/"//g' | sed 's/,//'`
+        echo "APP_IMAGE_TAG=ghcr.io/${{ github.repository }}${branch}:pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "APP_VERSION_IMAGE_TAG=ghcr.io/${{ github.repository }}${branch}_version:pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "BRANCH=${branch}" >> $GITHUB_ENV
+        echo "BUILD_DATE=$(date -u +"%Y-%m-%dT%H:%M:%SZ")" >> $GITHUB_ENV
+        echo "NARRATIVE_GIT_HASH=${narrative_git_hash}" >> $GITHUB_ENV
+        echo "NARRATIVE_VERSION=${narrative_version}" >> $GITHUB_ENV
+        echo "PR=pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "VCS_REF=$(echo "${{ github.event.pull_request.head.sha }}" | cut -c -7)" >> $GITHUB_ENV
```

Comment:
set up all our favourite environment variables... this chunk of code will start to look rather familiar